projects/gateway2/translator/httproute/gateway_http_route_translator.go

```diff
@@ -372,20 +365,30 @@ func setRouteAction(
 			}
 		}
 
+		obj, err := gwroute.GetBackendForRef(backendRef.BackendObjectReference)
 		fromPlugin := false
-		for _, bp := range pluginRegistry.GetBackendPlugins() {
-			if dest, ok := bp.ApplyBackendPlugin(obj, backendRef.BackendObjectReference); ok {
-				fromPlugin = true
-				weightedDestinations = append(weightedDestinations, &v1.WeightedDestination{
-					Destination: dest,
-					Weight:      weight,
-				})
-				break
+		if err == nil {
```

Comment:
We specifically do not want to. If we call ProcessBackendRef for some type that only a plugin knows about, we'll get a report about unsupported types when that isn't true.

Reply:
Understood. I think the `fromPlugin` var existed between the func call and the error handling in the previous commit. My ^ comment is null since you implemented https://github.com/solo-io/gloo/pull/10361#discussion_r1852969336.